substrate-node/pallets/pallet-tfgrid/src/grid_migration.rs

```diff
@@ -243,15 +243,35 @@ pub fn migrate_farms<T: Config>() -> frame_support::weights::Weight {
             }
         }
 
-        let replaced_farm_name = farm::replace_farm_name_spaces_with_underscores(&farm.name);
-        let name = match <T as Config>::FarmName::try_from(replaced_farm_name) {
+        let mut farm_name = farm.name;
+        let mut truncated = false;
+        if farm_name.len() > 40 {
+            info!("farm name length exceeded, removing from farm id by name map");
+            FarmIdByName::<T>::remove(&farm_name);
```

Comment:
This remove can be removed, and on line 257, the check switch the  `&& !truncated` to `|| truncated`. Side effects (such as modifying storage) should avoid repetition as much as possible. In fact the code here should probably be changed in its entirety such that all modifications on farm_name are done first, and then later there is just 1 block which checks if farm_name changed at all and if so replaces the storage map entry.